app/controllers/stats/data/backend_apis_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class Stats::Data::BackendApisController < Stats::Data::BaseController
+  before_action :set_source
+
+  ##~ sapi = source2swagger.namespace("Analytics API")
+  ##
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/stats/backend_apis/{backend_api_id}/usage.{format}"
+  #
+  ##~ op            = e.operations.add
+  ##~ op.httpMethod = "GET"
+  ##~ op.summary    = "Backend Traffic by Metric"
+  ##~ op.description = "Returns the traffic data of a given metric (or method) of a backend."
+  ##~ op.group = "backend_api_ops"
+  #
+  ##~ op.parameters.add @parameter_format
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_backend_api_id_by_id_name
+  ##~ op.parameters.add @parameter_backend_api_metric_name
+  ##~ op.parameters.add @parameter_since
+  ##~ op.parameters.add @parameter_period
+  ##~ op.parameters.add @parameter_until
+  ##~ op.parameters.add @parameter_granularity
+  ##~ op.parameters.add @parameter_timezone
+  ##~ op.parameters.add @parameter_skip_change
+  #
+
+  def summary
+    super
+  end
+
+  private
+
+  def set_source
+    backend_apis = current_account.backend_apis.accessible
+    @backend_api = backend_apis.find(params[:backend_api_id])
+    @source  = Stats::BackendApi.new(@backend_api)
+
+    authorize!(:show, @backend_api) if current_user
+  rescue ActiveRecord::RecordNotFound
+    render_error 'Backend not found', status: :not_found
```

Comment:
If this is removed, and I add this test in `test/integration/stats/data/backend_apis_controller_test.rb`:

```ruby
test 'when the backend api does not exist' do
    params = { metric_name: metric.system_name, period: 'day', since: Time.utc(2020, 3, 19).to_date, timezone: ActiveSupport::TimeZone['UTC'].name, skip_change: false }
    get usage_stats_data_backend_apis_path(backend_api_id: backend_api.id + 20, format: :json, access_token: access_token.value), params

    assert_response :not_found
    pp response.body
end
```
It works perfectly well and the response is just like in the other APIs: `"{\"status\":\"Not found\"}"`

I guess this custom error message is for the UI but I do not think that it is necessary... if it does a request to backend_apis and the response status is 404, it definitely means that the Backend is not found.